README: Make it more beginner-friendly

n0toose opened the pull request:
In the beginning, I found myself quite lost as I was not totally sure how to use Uhyve. This change introduces some additional background information and pointers for people looking to hack on the source code.

- Make "code blocks" more consistent
- Make some wording more consistent
- Additional Markdown magic
- Cut the length of some sentences
- Add `cargo run` method for developers
- Introduce GitHub-styled admonition boxes

jounathaen commented on README.md:
```diff
@@ -30,25 +29,27 @@ $ cargo install --locked uhyve
 
-Apple's *Command Line Tools* must be installed.
-The following terminal command installs these tools *without* Apple's IDE Xcode:
+You must install Apple's [Xcode Command Line Tools](https://developer.apple.com/xcode/resources) using the following command:
```
This sounds like any other way of installing the tools will inevitably fail

jounathaen commented on README.md:
````diff
@@ -96,19 +97,37 @@ The file `app.entitlements` must have following content:
 ```
 
-For further details have a look at [Apple's documentation](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_hypervisor).
+Please consult [Apple's Documentation](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_hypervisor) for more information.
````
for _further_ information

jounathaen commented on README.md:
````diff
@@ -96,19 +97,37 @@ The file `app.entitlements` must have following content:
+cargo run -- -v data/x86_64/hello_world
+cargo run -- -v data/x86_64/hello_c
+```
+
````
This rather should be moved to some kind of Wiki or a later section in the document

jounathaen commented on README.md:
````diff
@@ -139,6 +162,30 @@ The repository [hermit-rs](https://github.com/hermitcore/hermit-rs) provides [ex
 
 ![Debugging Hermit apps](img/vs_code.png)
 
+## Networking
+
+> [!CAUTION]
+> Network support is currently unstable and tested only on Linux.
+
+If you require uhyve to create its own virtual ethernet interface, you will need to provide it with the `CAP_NET_ADMIN` capability:
+
+```sh
+# as root
+setcap cap_net_admin+ep /path/to/uhyve # ./target/debug/uhyve
+```
+
+~~You can set the pre-created tap device name via an environment variable `TAP`~~
+
+Currently, the device is hard-coded with the name `tap10`. You will need to create the device and connect it to a bridge (such as virbr0):
+
+```sh
+ip tuntap add tap10 mode tap user "$(whoami)"
+ip link set tap10 master virbr0
+ip link set dev tap10 up
+```
+
+And, if desired, set the IP address and gateway of your RustyHermit instance via `HERMIT_IP` and `HERMIT_GATEWAY`.
+
````
This is nice, but the feature is not yet in the `main` branch

jounathaen reviewed (commented):
Very nice improvements. I've added some minor comments

n0toose commented on README.md:
```diff
@@ -7,20 +7,19 @@
 
-An installation of the Rust toolchain is required.
-Please visit the [Rust website](https://www.rust-lang.org/) and follow the installation instructions.
+1. Install the Rust toolchain using your preferred package manager or the [Rust Team's Installation Instructions](https://www.rust-lang.org/tools/install).
```
I'd gladly switch it back and see what you mean as a person who has also had major issues in the past (such as with `rustup`), but most people in my circle (NixOS, Arch Linux, Debian-based distributions, Fedora) and myself depend on package managers.

I'll rephrase so as to not make it sound like an explicit instruction.

mkroening replied:
I'd also strongly recommend rustup. It's a hard requirement for Hermit, even.

n0toose commented:
I addressed the review comments and made some additional changes to make them "fit" well, such as adding clearer headers and separating the content better - this introduces some new changes without changing the substance.

Perhaps it would be better if a `CONTRIBUTING.md` file were to be written in the future instead of cramming everything in the README file, but it should be adequate enough for now.

mkroening commented on README.md:
```diff
@@ -7,20 +7,25 @@
+1. Install the Rust toolchain.
 
-Install Uhyve with
+> [NOTE]
```
```suggestion
> [!NOTE]
```

jounathaen replied:
Do we need to highlight this as *note* at all? I think it is just a minor thing

n0toose replied:
gahhhhhh, sorry for the misses! i'll adjust this in a sec

jounathaen commented on README.md:
```diff
@@ -119,13 +131,48 @@ By default, the loader initializes a system with one core and 512 MiB RAM.
+
+### Debugging Hermit apps
+
+> [!WARNING]
+> This feature is unstable.
```
Final nitpicks: Please remove this, this feature is as unstable as everything else :grin:

jounathaen commented on README.md:
```diff
@@ -119,13 +131,48 @@ By default, the loader initializes a system with one core and 512 MiB RAM.
+
+### Testing Hermit apps using `cargo run`
+
+As mentioned above, the Uhyve repository ships some binaries that can be used for testing purposes.
+
+If you are hacking on Uhyve's source code, you can also use `cargo run` to immediately test your changes:
```
```suggestion
As mentioned above, the Uhyve repository ships some binaries that can be used for testing purposes:
```

jounathaen reviewed (approved):
Final Nitpicks: I'm overall quite happy with the new version!

jounathaen commented:
Oh, and I'd stash these changes into a single commit...

n0toose commented:
Wait, one more rebase!

jounathaen commented on README.md:
````diff
@@ -33,59 +32,62 @@ To check if your system supports virtualization, you can use the following comma
-```sh
-sysctl kern.hv_support
-```
+> [!NOTE]
+> To verify if your processor is able to support this framework, run the following in your Terminal:
+>
+> ```sh
+> sysctl kern.hv_support
+> ```
````
Oh, didn't see this earlier. I don't think this needs extra highlighting with a note. Especially, since the following line is also not hightlighted
